enhancements/sig-architecture/225-placement-score-visibility/README.md

```diff
@@ -0,0 +1,173 @@
+# Placement Score Visibility
+
+## Release Signoff Checklist
+
+- [] Enhancement is `provisional`
+- [] Design details are appropriately documented from clear requirements
+- [] Test plan is defined
+- [] Graduation criteria for dev preview, tech preview, GA
+- [] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal extends existing Placement mechanisms to surface the scores of each cluster
+in PlacementDecision so that users (or future OCM features) can make advanced scheduling decisions
+based on the scored state of clusters.
+
+In addition, it adds support for different sort strategies on the decisions generated by a Placement.
+By default, decisions will continue to be sorted by "Name". An additional sort strategy "Score" will
+enable sorting decisions numerically in descending order by prioritizer score.
+
+## Motivation
+
+Placement scores are a powerful feature of OCM for scheduling workloads, but there is currently
+no reliable way for users to see what scores were assigned to clusters for a placement. Scores are
+reported in events, but these are ephemeral. Since PlacementDecisions are orderd alphanumerically
+there is no good way to see the relative scores of clusters in a decision other than recomputing
+them for each cluster in the decision group(s).
+
+A user may wish to use Placement to find all clusters that match some conditions, but then pick
+only the top N clusters for their workload. While this could be accomplished by setting
+`numberOfClusters: N` on the placement, this means a new placement must be created for every value
+of N. Another use-case would be to distribute workloads out to selected clusters randomly weighted
+by their scores. There is currently no way to do this in OCM without recomputing scores for each cluster
+listed in the PlacementDecisions. Displaying scores in an easy to consume manner gives more flexibility
+to the user, and enables a single Placement to be reused for many different scheduling decisions
+(reducing API pressure on the hub).
+
+## Proposal
+
+Add `SortBy` to `PlacementSpec` to configure the sort order for PlacementDecision. This field
+is an enum with values "Name" and "Score". Add `Score` to `ClusterDecision` to reflect the
+latest score of each cluster in the decision.
+
+### Design Details
+
+#### API change
+
+Field for sorting decisions
+
+```go
+type PlacementSpec struct {
+  ...
+
+  // SortBy sets the sort order for decisions.
+  // It can be "Name", "Score", or "", where "" is "Name" by default.
+  // If sortBy is "Name", decisions will be orderd alphanumerically by cluster name
+  // If sortBy is "Score", decisions will be ordered numerically in descending order by score,
```

Comment:
As mentioned above, need to notice user that when sortBy is "Score", the placement decision will trigger an update when score changes. Since today we didn't control the frequency of how user update the AddonPlacmentScore, that means if the score changed too frequently, the decision result will be updated frequently also.

Reply:
Defining the time (eg. every 1 minutes) to reschedule the placement might be a way to avoid the pressure on API server, but the above discussed, the update of the score is uncontrollable, so perhaps the best way for now is just add some notice to the user.

Reply:
it might make sense to define a rateLimit, such as MaxUpdatePerMinute